contracts/ZKPVerifier.sol

```diff
@@ -7,7 +7,15 @@ import "./lib/GenesisUtils.sol";
 import "./interfaces/ICircuitValidator.sol";
 import "./interfaces/IZKPVerifier.sol";
 
+interface ISpongePoseidon {
+    function hash(uint256[] calldata values) external view returns (uint256);
+
+    function hash4(uint256[4] calldata values) external view returns (uint256);
+}
```

Comment:
This interface looks like an attempt of multi-option Poseidon definition. If so, the name could be better

Reply:
@AndriianChestnykh could you give an example?